File: python/栈/682. 棒球比赛.py
from typing import List
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        res = []
        for op in ops:
            if op == "C":
                res.pop()
            elif op == "D":
                res.append(res[-1]*2)
            elif op == "+":
                res.append(res[-1]+res[-2])
            else:
                res.append(int(op))
        return sum(res)
s = Solution()
ops = ["5","-2","4","C","D","9","+","+"]
print(s.calPoints(ops))

File: go/stack/682. 棒球比赛.go
package stack

import "strconv"

func calPoints(ops []string) int {
	var res []int
	for _, v := range ops {
		if v == "C" {
			res = res[:len(res)-1]
		} else if v == "D" {
			res = append(res, res[len(res)-1]*2)
		} else if v == "+" {
			res = append(res, res[len(res)-1]+res[len(res)-2])
		} else {
			int_v, _ := strconv.Atoi(v)
			res = append(res, int_v)
		}
	}
	var sum int
	for _, v := range res {
		sum += v
	}
	return sum
}
